### PR TITLE
perf: add default d3-flame-graph templates for flamegraph script

### PR DESCRIPTION
**Pull request by aaronjheng**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/kernel/perf/default.nix [hunk 1]
```diff
@@ -1,11 +1,58 @@
...
+, withGtk ? false
+, gtk2
+, withZstd ? true
+, zstd
+, withLibcap ? true
+, libcap
 }:
+let
+  d3-flame-graph-templates = stdenv.mkDerivation rec {
+    name = "d3-flame-graph-templates";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/kernel/perf/default.nix:43]:
```suggestion
    pname = "d3-flame-graph-templates";
```

**Comment by SuperSandro2000** [comment]
@ofborg build cargo-flamegraph
@ofborg build linuxPackages.perf
@ofborg build linuxKernel.packages.linux_6_0.perf
@ofborg build linuxKernel.packages.linux_5_15.perf

**Comment by trofi** [comment]
The change broke `linuxPackages_5_4.perf` build as `flamegraph.py` was added only in `linux-5.10`:

```
substitute(): ERROR: file 'tools/perf/scripts/python/flamegraph.py' does not exist
```

**Comment by aaronjheng** [comment]
@tdroxler https://github.com/NixOS/nixpkgs/pull/201127

**Comment by aaronjheng** [comment]
Sorry for the noise @tdroxler. I mean @trofi.

**Pull request merged**